src/parser/ast/mod.rs

```diff
@@ -395,15 +395,17 @@ pub enum Expression {
         question_arrow: Span, // `?->`
         property: Box<Self>,  // `bar`
     },
-    // `foo()::bar`
+    // `foo()::$bar`
     StaticPropertyFetch {
         target: Box<Self>,   // `foo()`
         double_colon: Span,  // `::`
-        property: Box<Self>, // `bar`
+        property: Box<Self>, // `$bar`
     },
-    ConstFetch {
-        target: Box<Self>,
-        constant: SimpleIdentifier,
+    // `foo()::bar` or `foo()::{$name}`
+    ConstantFetch {
+        target: Box<Self>,   // `foo()`
+        double_colon: Span,  // `::`
+        constant: Box<Self>, // `bar`
```

Comment:
we want to improve this in the future, it's `DynamicIdentifier` or `SimpleIdentifier`, not a full expression.

Reply:
fixed.